bumble/hid.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2021-2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -----------------------------------------------------------------------------
+# Imports
+# -----------------------------------------------------------------------------
+from __future__ import annotations
+import logging
+import asyncio
+
+from pyee import EventEmitter
+from typing import Optional, Tuple, Callable, Dict, Union
+from .device import Device, Connection
+
+from . import core, l2cap  # type: ignore
+from .colors import color  # type: ignore
+from .core import BT_BR_EDR_TRANSPORT, InvalidStateError, ProtocolError  # type: ignore
+
+# -----------------------------------------------------------------------------
+# Logging
+# -----------------------------------------------------------------------------
+logger = logging.getLogger(__name__)
+
+
+# -----------------------------------------------------------------------------
+# Constants
+# -----------------------------------------------------------------------------
+# fmt: off
+
+HID_CONTROL_PSM            = 0x0011
+HID_INTERRUPT_PSM          = 0x0013
+
+# HIDP message types
+HID_HANDSHAKE           = 0x00
+HID_CONTROL             = 0x01
+HID_GET_REPORT          = 0x04
+HID_SET_REPORT          = 0x05
+HID_GET_PROTOCOL        = 0x06
+HID_SET_PROTOCOL        = 0x07
+HID_DATA                = 0x0A
+
+# Report types
+HID_OTHER_REPORT        = 0x00
+HID_INPUT_REPORT        = 0x01
+HID_OUTPUT_REPORT       = 0x02
+HID_FEATURE_REPORT      = 0x03
+
+# Handshake parameters
+HANDSHAKE_SUCCESSFUL              = 0x00
+HANDSHAKE_NOT_READY               = 0x01
+HANDSHAKE_ERR_INVALID_REPORT_ID   = 0x02
+HANDSHAKE_ERR_UNSUPPORTED_REQUEST = 0x03
+HANDSHAKE_ERR_UNKNOWN             = 0x0E
+HANDSHAKE_ERR_FATAL               = 0x0F
+
+# Protocol modes
+HID_BOOT_PROTOCOL_MODE      = 0x00
+HID_REPORT_PROTOCOL_MODE    = 0x01
+
+# Control Operations
+HID_SUSPEND              = 0x03
+HID_EXIT_SUSPEND         = 0x04
+HID_VIRTUAL_CABLE_UNPLUG = 0x05
+
+
+class HIDPacket():
```

Comment:
This class seems to be a good candidate for [`dataclass`](https://docs.python.org/3/library/dataclasses.html)